Release tests fail because bwc artifacts are not generated

The release tests are failing on CI with the following error (this specific output was taken from https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+release-tests/471/console but there are other build runs failing too on master and 6.x) :

```
00:51:09 FAILURE: Build failed with an exception.
00:51:09 
00:51:09 
00:51:09 * Where:
00:51:09 Deprecated Gradle features were used in this build, making it incompatible with Gradle 5.0.
00:51:09 Build file '/var/lib/jenkins/workspace/elastic+elasticsearch+master+release-tests/distribution/bwc/build.gradle' line: 159
00:51:09 See https://docs.gradle.org/4.5/userguide/command_line_interface.html#sec:command_line_warnings
00:51:09 
00:51:09 2244 actionable tasks: 2193 executed, 51 up-to-date
00:51:09 * What went wrong:
00:51:09 Execution failed for task ':distribution:bwc:next-bugfix-snapshot:buildBwcVersion'.
00:51:09 > Building bwc version didn't generate expected files [/var/lib/jenkins/workspace/elastic+elasticsearch+master+release-tests/distribution/bwc/next-bugfix-snapshot/build/bwc/checkout-6.2/distribution/deb/build/distributions/elasticsearch-6.2.2-SNAPSHOT.deb, /var/lib/jenkins/workspace/elastic+elasticsearch+master+release-tests/distribution/bwc/next-bugfix-snapshot/build/bwc/checkout-6.2/distribution/rpm/build/distributions/elasticsearch-6.2.2-SNAPSHOT.rpm, /var/lib/jenkins/workspace/elastic+elasticsearch+master+release-tests/distribution/bwc/next-bugfix-snapshot/build/bwc/checkout-6.2/distribution/zip/build/distributions/elasticsearch-6.2.2-SNAPSHOT.zip]
```

@hub-cap could these failures be related to your bwc changes in the build last week?

--- a/distribution/bwc/build.gradle
+++ b/distribution/bwc/build.gradle
@@ -139,7 +139,7 @@ subprojects {
     } else {
       executable new File(checkoutDir, 'gradlew').toString()
     }
-    args ":distribution:deb:assemble", ":distribution:rpm:assemble", ":distribution:zip:assemble", "-Dbuild.snapshot=${System.getProperty('build.snapshot') ?: 'true'}"
+    args ":distribution:deb:assemble", ":distribution:rpm:assemble", ":distribution:zip:assemble", "-Dbuild.snapshot=true"
     final LogLevel logLevel = gradle.startParameter.logLevel
     if ([LogLevel.QUIET, LogLevel.WARN, LogLevel.INFO, LogLevel.DEBUG].contains(logLevel)) {
       args "--${logLevel.name().toLowerCase(Locale.ENGLISH)}"

